src/Tests/NodeIntegrationTest.php

```diff
@@ -255,4 +256,84 @@ public function testDataSet() {
     $this->assertNotNull($node->id(), 'Node ID is set, which means that the node has been auto-saved.');
   }
 
+  /**
+   * Tests that a required context value that is NULL throws an exception.
+   */
+  public function testRequiredNullValue() {
```

Comment:
it seems wrong to have that in NodeIntegrationTests, can we move this to engine tests?